### ISSUE TITLE
DemangledFunction: IllegalArgumentException: External symbol may not be null

### ISSUE THREAD

**Issue by ning1022** [NationalSecurityAgency/ghidra#3619]
How to solve this ghidra  error?
```
(These messages are also written to the application log file)

Demangler GNU> Apply failure (DemangledFunction: IllegalArgumentException: External symbol may not be null)
	_ZNSt8xxxx_xxx4xxxD1xxxx
```

IDA can parse this file。

**Comment by dragonmacher** [comment]
This likely an easy fix, but we may need the binary in order to reproduce the issue.   

The error is being created because the Demangler is trying to create an external function for the address of the symbol you are showing, but  it cannot find a primary symbol at that address.  The error does not really affect the disassembly, but it may mean that those functions do not get created.  You can go to each symbol listed in the log and manually create a function there, if one does not already exist.

**Comment by ning1022** [comment]
"You can go to each symbol listed in the log and manually create a function" Where is the directory of this log file?

**Comment by astrelsky** [comment]
> 
> How to solve this ghidra  error?
> ```
> (These messages are also written to the application log file)
> 
> Demangler GNU> Apply failure (DemangledFunction: IllegalArgumentException: External symbol may not be null)
> 	_ZNSt8xxxx_xxx4xxxD1xxxx
> ```
> 
> IDA can parse this file。
> 
> 
> 

Did you mean `_ZNSt8xxxx_xxx4xxxxD1xxxx`?

**Comment by dragonmacher** [comment]
> "You can go to each symbol listed in the log and manually create a function" Where is the directory of this log file?

The application.log file lives in `{user.home}/.ghidra/.ghidra_{version}/`.

It may be easier to use the `Symbol Table` view to show all symbols and then filter with the mangled prefix that matches this error case (you can use a regex filter if that helps).

**Comment by ning1022** [comment]
@dragonmacher  When I opened this file( application.log), I find  the following error message:
```
2021-11-22 22:20:52 DEBUG (ToolTaskManager) Exec Task Auto Analysis  
2021-11-22 22:20:55 INFO  (DWARFImportSummary) DWARF import - total elapsed: 2414ms  
2021-11-22 22:20:55 INFO  (DWARFImportSummary) DWARF data type import - elapsed: 1582ms  
2021-11-22 22:20:55 INFO  (DWARFImportSummary) DWARF func & symbol import - elapsed: 496ms  
2021-11-22 22:20:55 INFO  (DWARFImportSummary) DWARF types imported: 660  
2021-11-22 22:20:55 INFO  (DWARFImportSummary) DWARF function signatures added: 455  
2021-11-22 22:20:55 ERROR (DWARFImportSummary) DWARF variable definitions that failed because the data type was too large for the defined register location: 1  
2021-11-22 22:20:55 ERROR (DWARFImportSummary) DWARF variable definitions that failed because they are computed pseudo variables: 6  
2021-11-22 22:20:55 ERROR (DWARFImportSummary) DWARF expression failed to read: 1  
2021-11-22 22:21:07 INFO  (AutoAnalysisManager) Analysis Log Messages
Demangler GNU> Apply failure (DemangledFunction: IllegalArgumentException: External symbol may not be null)
	_xxxx_xxxxx
```

**Comment by dragonmacher** [comment]
I think this either has been fixed or is in the process of being fixed.  The fix should make it into 10.1.

**Comment by ning1022** [comment]
Yes,  this   has been fixed  the 10.1-BETA release .
